core/state_processor.go

```diff
@@ -418,6 +425,24 @@ func applyTransaction(msg types.Message, config *params.ChainConfig, bc ChainCon
 	if err != nil {
 		return nil, err
 	}
+
+	primeTerminus := bc.GetPrimeTerminus(parent)
+	if result != nil && len(result.Etxs) > 0 {
+		for _, etx := range result.Etxs {
+			if !bc.IsSliceSetToReceiveEtx(primeTerminus, *etx.To().Location()) {
+				result, err = &ExecutionResult{
+					UsedGas:      params.TxGas,
```

Comment:
You should charge them the previously used gas because the execution was technically done. I also think in the future this failure should be done in CreateETX but not necessary now

Reply:
using the result.UsedGas